AndroidSDK/src/com/leanplum/internal/Request.java

```diff
@@ -390,71 +382,84 @@ private void sendNow() {
     Util.executeAsyncTask(true, new AsyncTask<Void, Void, Void>() {
       @Override
       protected Void doInBackground(Void... params) {
-        List<Map<String, Object>> requestsToSend = getUnsentRequests();
-        if (requestsToSend.isEmpty()) {
-          return null;
-        }
+        return sendRequests();
+      }
+    });
+  }
 
-        final Map<String, Object> multiRequestArgs = new HashMap<>();
-        multiRequestArgs.put(Constants.Params.DATA, jsonEncodeUnsentRequests(requestsToSend));
-        multiRequestArgs.put(Constants.Params.SDK_VERSION, Constants.LEANPLUM_VERSION);
-        multiRequestArgs.put(Constants.Params.ACTION, Constants.Methods.MULTI);
-        multiRequestArgs.put(Constants.Params.TIME, Double.toString(new Date().getTime() / 1000.0));
-        if (!Request.attachApiKeys(multiRequestArgs)) {
-          return null;
-        }
+  private Void sendRequests() {
+    List<Map<String, Object>> requestsToSend = getUnsentRequests();
+    if (requestsToSend.isEmpty()) {
+      return null;
+    }
 
-        JSONObject result = null;
-        HttpURLConnection op = null;
-        try {
-          try {
-            op = Util.operation(
-                Constants.API_HOST_NAME,
-                Constants.API_SERVLET,
-                multiRequestArgs,
-                httpMethod,
-                Constants.API_SSL,
-                Constants.NETWORK_TIMEOUT_SECONDS);
+    final Map<String, Object> multiRequestArgs = new HashMap<>();
+    multiRequestArgs.put(Constants.Params.DATA, jsonEncodeUnsentRequests(requestsToSend));
+    multiRequestArgs.put(Constants.Params.SDK_VERSION, Constants.LEANPLUM_VERSION);
+    multiRequestArgs.put(Constants.Params.ACTION, Constants.Methods.MULTI);
+    multiRequestArgs.put(Constants.Params.TIME, Double.toString(new Date().getTime() / 1000.0));
+    if (!Request.attachApiKeys(multiRequestArgs)) {
+      return null;
+    }
+
+    JSONObject result = null;
+    HttpURLConnection op = null;
+    try {
+      try {
+        op = Util.operation(
+            Constants.API_HOST_NAME,
+            Constants.API_SERVLET,
+            multiRequestArgs,
+            httpMethod,
+            Constants.API_SSL,
+            Constants.NETWORK_TIMEOUT_SECONDS);
 
-            result = Util.getJsonResponse(op);
-            int statusCode = op.getResponseCode();
+        result = Util.getJsonResponse(op);
+        int statusCode = op.getResponseCode();
 
-            Exception errorException = null;
-            if (statusCode == 200) {
-              deleteSentRequests(requestsToSend.size());
-            } else if (statusCode >= 400) {
-              errorException = new Exception("HTTP error " + statusCode);
+        Exception errorException = null;
+        if (statusCode >= 200 && statusCode <= 299) {
```

Comment:
Hm, I tried to refactor it in the way I thought, but I ended up refactoring almost everything... I think we should refactor/rename after we have this completed, and we have all tests in place....